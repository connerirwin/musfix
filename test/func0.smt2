; Parsing of uniterpreted functions
(qualif Eq ((v @a)(z @b)) (= v z))

(declare-fun bar (Int) Bool)

(wf $k0 ((v0 Int)(v1 Int)))

; Various Errors
; (qualif QQ ((b Bool)) (= (bar b) True))
; (qualif QQ ((i Int)) (= (bar i) 1))

(constraint
  (forall ((v2 Int)(v3 Int))
          (=> ($k0 v2 v3) (= (bar v2)(bar v3)))))
