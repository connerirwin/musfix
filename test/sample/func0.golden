
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/sample/func0.msmt
Qualifier "Eq" [Var (VarS "a") "v",Var (VarS "b") "z"] (Binary Eq (Var (VarS "a") "v") (Var (VarS "b") "z"))
UninterpFunction "bar" [IntS] BoolS
WFConstraint "$k0" [Var IntS "v0",Var IntS "v1"]
HornConstraint [Var IntS "v2",Var IntS "v3"] (Binary Implies (Unknown (fromList [("v0",Var IntS "v2"),("v1",Var IntS "v3")]) "$k0") (Binary Eq (Func BoolS "bar" [Var IntS "v2"]) (Func BoolS "bar" [Var IntS "v3"])))

QMAP
--------
fromList [("$k0",QSpace {_qualifiers = [Binary Eq (Var IntS "v0") (Var IntS "v1"),Binary Eq (Var IntS "v1") (Var IntS "v0")], _maxCount = 2})]

Candidates
--------
[Candidate {solution = fromList [("$k0",fromList [Binary Eq (Var IntS "v1") (Var IntS "v0")])], validConstraints = fromList [Binary Implies (Unknown (fromList [("v0",Var IntS "v2"),("v1",Var IntS "v3")]) "$k0") (Binary Eq (Func BoolS "bar" [Var IntS "v2"]) (Func BoolS "bar" [Var IntS "v3"]))], invalidConstraints = fromList [], label = "0.0"},Candidate {solution = fromList [("$k0",fromList [Binary Eq (Var IntS "v0") (Var IntS "v1")])], validConstraints = fromList [Binary Implies (Unknown (fromList [("v0",Var IntS "v2"),("v1",Var IntS "v3")]) "$k0") (Binary Eq (Func BoolS "bar" [Var IntS "v2"]) (Func BoolS "bar" [Var IntS "v3"]))], invalidConstraints = fromList [], label = "0.1"}]


Final candidates: 
0.0: [$k0 -> {v1 [37m==[0m v0}] (1 0)
0.1: [$k0 -> {v0 [37m==[0m v1}] (1 0)

